packages/harmony/src/components/input/TextInput/TextInput.mdx

```diff
@@ -0,0 +1,240 @@
+import {
+  Canvas,
+  Meta,
+  Unstyled,
+  Title,
+  Subtitle,
+  Description,
+  Controls,
+  Primary
+} from '@storybook/blocks'
+
+import { Flex } from 'components'
+import { IconSearch } from 'icons'
+import { ComponentRules, RelatedComponents } from 'storybook/components'
+
+import { TextInput } from './TextInput'
+import * as TextInputStories from './TextInput.stories'
+import { TextInputSize } from './types'
+
+<Meta of={TextInputStories} />
+
+<Title />
+
+- [Overview](#overview)
+- [Props](#props)
+- [Configuration options](#configuration-options)
+- [Use cases and examples](#use-cases-and-examples)
+- [Do's and dont's](#dos-and-donts)
+- [Related Components](#related-components)
+
+## Overview
+
+<Subtitle />
+<Description />
+<Primary />
+
+## Props
+
+<Controls />
+
+## Configuration Options
+
+### Sizes
+
+<Canvas of={TextInputStories.Sizes} />
+
+### States
+
+<Canvas of={TextInputStories.DefaultStates} />
+<Canvas of={TextInputStories.SmallStates} />
```

Comment:
do we need both versions of this? the large and small ones seem to behave the same wrt changing states

Reply:
matching designs here, it was split by large and small in two canvases